### ISSUE TITLE
Pagination code not accessible

### ISSUE THREAD

**Issue by guermanc** [UKHomeOffice/design-system#345]
Pagination component doesn't currently follow accessibility best practice:
- group pagination in a `<nav>` element, provide a `role=navigation` and `aria-label` to expose role to assistive technology
- voice the current page with `aria-current="page"`
- indicate clearly that each link number corresponds to a page with `aria-label`

See https://designsystem.digital.gov/components/pagination/ and https://www.a11ymatters.com/pattern/pagination/

**Comment by MikeDThorpe** [comment]
I'll pick this accessibility work up together with issue #340 as part of the branch [pagination-redesign](https://github.com/UKHomeOffice/design-system/tree/pagination-redesign)
